src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1189,19 +1194,46 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 Icon icon = chooser.getIcon(file);
                 setIcon(icon);
 
-            } else if (value instanceof Long) {
-                long len = ((Long) value) / 1024L;
+            } else if (value instanceof Long len) {
                 if (listViewWindowsStyle) {
-                    text = MessageFormat.format(kiloByteString, len + 1);
-                } else if (len < 1024L) {
-                    text = MessageFormat.format(kiloByteString, (len == 0L) ? 1L : len);
+                    updateMessageFormatPattern(kiloByteString, 1);
+                    if (len == 0) {
+                        objs[0] = Double.valueOf(0.0);
+                    } else if (len > 0 && len < 100L) {
+                        objs[0] = Double.valueOf(0.1);
+                    } else {
+                        double kbVal = formatToDoubleValue(len);
+                        objs[0] = Double.valueOf(kbVal);
```

Comment:
You can assign `double` directly without explicit boxing with `Double.valueOf`.

```suggestion
                        objs[0] = 0.0;
                    } else if (len > 0 && len < 100L) {
                        objs[0] = 0.1;
                    } else {
                        objs[0] = formatToDoubleValue(len);
```

IDE highlights each line with such usage.